sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/buildfeatures/BranchesAndPullRequestsParametersPreprocessor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jetbrains.buildserver.sonarplugin.buildfeatures;
+
+import java.util.Map;
+
+import org.springframework.util.StringUtils;
+
+import com.google.gson.JsonObject;
+
+import jetbrains.buildServer.messages.Status;
+import jetbrains.buildServer.serverSide.ParametersPreprocessor;
+import jetbrains.buildServer.serverSide.SRunningBuild;
+import jetbrains.buildServer.serverSide.buildLog.MessageAttrs;
+
+/**
+ * Provides SONARQUBE_SCANNER_PARAMS environment variable on build if feature activated
+ */
+public class BranchesAndPullRequestsParametersPreprocessor implements ParametersPreprocessor {
```

Comment:
I have hesitate with `BuildParametersProvider` extension ... but `SONARQUBE_SCANNER_PARAMS` sysenv should not really be editable ...